src/examples/encryption.ts

```diff
@@ -0,0 +1,41 @@
+import { Encryption, Encoding, PrivateKey, isReady, Circuit } from 'snarkyjs';
+
+await isReady;
+
+// generate keys
+let privateKey = PrivateKey.random();
+let publicKey = privateKey.toPublicKey();
+
+// message
+let message = 'This is a secret.';
+let messageFields = Encoding.Bijective.Fp.fromString(message);
+
+// encrypt
+let cipherText = Encryption.encrypt(messageFields, publicKey);
+
+// decrypt
+let decryptedFields = Encryption.decrypt(cipherText, privateKey);
+let decryptedMessage = Encoding.Bijective.Fp.toString(decryptedFields);
+
+if (decryptedMessage !== message) throw Error('decryption failed');
+console.log(`Original message: "${message}"`);
+console.log(`Recovered message: "${decryptedMessage}"`);
+
+// the same but in a checked computation
+
+await Circuit.runAndCheck(async () => {
+  // encrypt
+  let cipherText = Encryption.encrypt(messageFields, publicKey);
```

Comment:
probably want to do the encryption outside of the circuit

Reply:
I think it can be useful though - what if we want to store a secret on-chain, and prove how it has been computed?

Reply:
yeah true, I guess only the scalar.random() sounds like it should be done outside the circuit. I remember asking that question a while ago, if we should forbid this function from being run within a circuit